src/main/java/org/zalando/nakadi/validation/schema/ChangeVersionAndCreatedAtConstraint.java

```diff
@@ -0,0 +1,20 @@
+package org.zalando.nakadi.validation.schema;
+
+import org.zalando.nakadi.domain.EventType;
+
+import java.util.Optional;
+
+public class ChangeVersionAndCreatedAtConstraint implements SchemaEvolutionConstraint {
+
+    @Override
+    public Optional<SchemaEvolutionIncompatibility> validate(final EventType original, final EventType eventType) {
+        if (original.getSchema().equals(eventType.getSchema())) {
+            if (!original.getSchema().getVersion().equals(eventType.getSchema().getVersion())) {
+                return Optional.of(new SchemaEvolutionIncompatibility("changing schema version is not allowed"));
```

Comment:
@adyach I end up removing this class in my latest change. I introduced two new classes EventTypeBase and EventTypeSchemaBase which removes the need for this class, since it made it impossible for users to change attributes we don't want them to change.